saleor/graphql/discount/mutations.py

```diff
@@ -382,4 +403,83 @@ def perform_mutation(cls, _root, info, **data):
             info, data.get("id"), only_type=Sale, field="sale_id"
         )
         cls.remove_catalogues_from_node(sale, data.get("input"))
-        return SaleRemoveCatalogues(sale=sale)
+        return SaleRemoveCatalogues(sale=ChannelContext(node=sale, channel_slug=None))
+
+
+class SaleChannelListingAddInput(graphene.InputObjectType):
+    channel_id = graphene.ID(required=True, description="ID of a channel.")
+    discount_value = PositiveDecimal(description="The value of the discount.")
+
+
+class SaleChannelListingInput(graphene.InputObjectType):
+    add_channels = graphene.List(
+        graphene.NonNull(SaleChannelListingAddInput),
+        description="List of channels to which the sale should be assigned.",
+        required=False,
+    )
+    remove_channels = graphene.List(
+        graphene.NonNull(graphene.ID),
+        description=("List of channels from which the sale should be unassigned."),
+        required=False,
+    )
+
+
+class SaleChannelListingUpdate(BaseChannelListingMutation):
+    sale = graphene.Field(Sale, description="An updated sale instance.")
+
+    class Arguments:
+        id = graphene.ID(required=True, description="ID of a sale to update.")
+        input = SaleChannelListingInput(
+            required=True,
+            description="Fields required to update sale channel listings.",
+        )
+
+    class Meta:
+        description = "Manage sale's availability in channels."
+        permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
+        error_type_class = DiscountError
+        error_type_field = "discount_errors"
+
+    @classmethod
+    def add_channels(cls, sale: "SaleModel", add_channels: List[Dict]):
+        for add_channel in add_channels:
+            defaults = {}
+            discount_value = add_channel.get("discount_value")
+            if discount_value:
+                defaults["discount_value"] = discount_value
```

Comment:
Lack of `validate_price_precision` and test for it. 
This validation is always required for any prices form inputs.